### PR TITLE
Add fib_hiroTakechi.py

### PR DESCRIPTION
**Pull request by HiroTestStudent**
Hello,

I worked on the fibonacci codes after class and added three codes to the codes that we worked on in-class on Tuesday.

Thanks,
Hiro

### CODE REVIEW

File: students/hirot_directory/fib_hiroTakechi.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	assert fib(0) == 0
+	assert fib(1) == 1
+	assert fib(2) == 1
+	assert fib(3) == 2
+	print(fib(10))
+
+# the codes that I worked on after class: 
+
+## code 1:
+def fibonacci(n, a=1, b=0):
```
Comment (PythonCHB) [inline students/hirot_directory/fib_hiroTakechi.py:27]: this is VERY slick! I've never seen this solution, but it's efficient and clean. Though kinda hard to follow the logic.

File: students/hirot_directory/fib_hiroTakechi.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return b if n < 1 else fibonacci(n - 1, a + b, a)
+
+print(fibonacci(7))
+
+
+## code 2:
+class Program():
```
Comment (PythonCHB) [inline students/hirot_directory/fib_hiroTakechi.py:34]: we'll get to classes in a couple weeks, but this is not an appropriate use case.

And it doesn't work -- the Fib method is calling a global Fib()function -- or it will fail if there isn't one.

**Review by PythonCHB — APPROVED** [review]

File: students/hirot_directory/fib_hiroTakechi.py (hunk 2 above)
Comment (HiroTestStudent) [reply students/hirot_directory/fib_hiroTakechi.py:34]: Hi Chris,

Thank you for the feedback. I just attended Maria's office hours and she also said we would cover the class and OOP in a few weeks. I just wanted to try it since Foundations of Python Programming class required to use their OOP template..

Thanks,
Hiro

File: students/hirot_directory/fib_hiroTakechi.py (hunk 1 above)
Comment (HiroTestStudent) [reply students/hirot_directory/fib_hiroTakechi.py:27]: Hi Chris,

I think this is from the recursion exercise (or homework) of Foundations of Algorithm and Data Structure, if I remember correctly. 

Thanks,
Hiro

Comment (PythonCHB) [reply students/hirot_directory/fib_hiroTakechi.py:27]: you could have claimed credit :-)
